### DISCUSSION TITLE
Possible optimization of the JS part of codebase

### DISCUSSION THREAD

**Discussion by AndreyDmitriev39r** [jdwilkin4/contrast-ratio-repo#80] [Ideas]
### General

I have some ideas for tickets, but before creating tickets I would like to get some feedback and also wait till  work on issues #37 #77 #79 will be finished: that way we can do more cleanup with less moves.

### Optimize if check for swatches dynamic

```
if (
    firstColor.length === 7 ||
    rgbRegex.test(firstColor) ||
    rgbaRegex.test(firstColor)
  ) {
    foregroundSwatch.style.backgroundColor = firstColor;
  }
  if (
    secondColor.length === 7 ||
    rgbRegex.test(secondColor) ||
    rgbaRegex.test(secondColor)
  ) {
```

These if checks are very similar, and they will be more verbose when we add more color formats. We can create some functions which returns boolean stuff, and use it in both checks. That way we can avoid some repetition.

### Decomposition of displayResult()

This function may become a big one. Maybe at some point we will need to split it a bit. Would like to hear ideas about this.

### innerHtml or innerText

We are using `innerHTML` to provide ratio now. I am not sure if there is a big difference between `innerHTML` and `innerText`. But. Do we need innerHTML, since we are rendering just text and nothing more?

**Comment by jdwilkin4** [comment]
> These if checks are very similar, and they will be more verbose when we add more color formats. We can create some functions 
> which returns boolean stuff, and use it in both checks. That way we can avoid some repetition.

Yeah I agree with that. 

> This function may become a big one. Maybe at some point we will need to split it a bit. Would like to hear ideas about this.

We could play around with creating some functions to cut down on the repetitiveness.
We could also see if a switch statement would work better in that case instead of a long if/else chain.

We could also look into extracting all of the helper functions into a separate helper function file and then importing it into the `index.js` file. That way the main `index.js` file will only be responsible for dealing with the display logic of the different color formats. 

> I am not sure if there is a big difference between innerHTML and innerText.
`innerText` sets the plain text between the opening and closing tags while `innerHTML` sets the content in HTML format.

**Comment by Okibe00** [comment]
>These if checks are very similar, and they will be more verbose when we add more color formats. We can create some functions which returns boolean stuff, and use it in both checks. That way we can avoid some repetition.


A low hanging solution for this may be to use a single regex to check if the input is correct.
something like

```
swatchBackground = (color1, color2) => {
  if(validInputRegex.test(color1)){
    foregroundSwatch.style.backgroundColor = color1;
  }
  if(validInputRegex.test(color2)){
    backgroundSwatch.style.backgroundColor = color2;  
  }
}
```
we can then call the function within display
swatchBackground(firstColor, secondColor)
